### DISCUSSION TITLE
How to build sorting query with RAND() as the second attribute

### DISCUSSION THREAD

**Discussion by d47081** [manticoresoftware/manticoresearch-php#176] [Q&A] [answered]
I need to randomize search results with same sorting attribute in results:

on my case it's web crawler, that scan older collected links first by timestamp (or with zero time value when row not indexed yet)

so crawler stuck on huge resource, and indexing only it documents in queue. The `RAND()` function could resolve this issue but seems it's not work and I can't debug the resulting query.

here is code construction to extend
https://github.com/YGGverse/Yo/blob/main/src/cli/document/crawl.php#L67

```
    $search = new \Manticoresearch\Search(
        $client
    );

    $search->setIndex(
        $config->manticore->index->document->name
    );

    $search->match(
        '*',
        'url'
    );

    $search->sort(
        'time',
        'asc'
    );
```

no success by appending this one for example
```
    $search->sort(
        'RAND()'
    );
```

My goal is the query like this but using `manticoresearch-php` API

```
SELECT * FROM index ORDER BY time ASC, RAND()
```

**Comment by manticoresearch** [comment, marked as answer]
Use the `expression()` function:

```php
<?php
$config = ['host'=>'127.0.0.1','port'=>9308];
$client = new \Manticoresearch\Client($config);
$index = $client->index('t');
$index->drop(true);
$index->create([
    'a' => ['type' => 'int'],
]);

$index->addDocument(['a' => 1]);
$index->addDocument(['a' => 1]);

print_r($index->search('')->expression('r', "rand()")->sort('a', 'asc')->sort('r', 'asc')->get());
```

**Reply by d47081** [reply]
Thanks! it work now
https://github.com/YGGverse/Yo/commit/33cc7789997b12f52d843010f18b4992f8fd463a
